src/v/cluster/self_test_rpc_types.h

```diff
@@ -386,6 +388,8 @@ struct get_status_response
     std::vector<self_test_result> results;
     self_test_stage stage{};
 
+    auto serde_fields() { return std::tie(id, status, stage, results); }
```

Comment:
Shouldn't this be 
` auto serde_fields() { return std::tie(id, status, results, stage); }`

to remain compatible with previous version that simply does serde in order of its fields?

Reply:
ya, but in my defense it looks like the ordering changed recently 😇

Reply:
Yup- previously a bug someone introduced here- which is exactly why this PR is going to be awesome.